### PR TITLE
Cleanup match after #8000. Also add proper nuking of enterprise profiles (related #8065)

### PR DESCRIPTION
**Pull request by lacostej**
Note that the cleanup reveals what appears to be a broken test

### CODE REVIEW

File: match/lib/match/nuke.rb [hunk 1]
```diff
@@ -170,21 +172,21 @@ def delete_files!
...
-      profile_type = Spaceship.provisioning_profile.development if type == :development
-
-      profile_type
+    def profile_type(prov_type)
+      {
+        appstore:    Spaceship.provisioning_profile.app_store,
+        development: Spaceship.provisioning_profile.development,
+        enterprise:  Spaceship.provisioning_profile.in_house,
+        adhoc:       Spaceship.provisioning_profile.ad_hoc
+      }[prov_type] ||= raise "Unknown prov. type #{prov_type}"
```
Comment (KrauseFx) [inline match/lib/match/nuke.rb]: Let's write `provisioning type`

**Review by hjanuschka — APPROVED** [review]

**Comment by hjanuschka** [comment]
@lacostej see felix last comment i think it slipped through

**Comment by lacostej** [comment]
@hjanuschka where? I think github was a bit slow to update the PR state.

**Comment by hjanuschka** [comment]
yep - seemed to be github cache or whatever.

**Pull request merged**